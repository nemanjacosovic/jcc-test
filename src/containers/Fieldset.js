import React, { Component } from 'react'
import PropTypes from 'prop-types'
import InputBox from '../components/InputBox'

class Fieldset extends Component {
	render() {
		const { showError, cssc, legend, type, name, inputs, cleanInput, onInputChange } = this.props
		let displayError

		if (showError) {
			displayError = (
				<div className='error-box' key={ cssc }>
					<p className='error-box-message'><span className='error-box-icon'></span>The field is required.</p>
				</div>
			)
		}

		return (
			<fieldset className={ cssc }>
				<legend>{ legend }</legend>
				<InputBox
					cssc			= { cssc }
					type    		= { type }
					name			= { name }
					inputs  		= { inputs }
					showError		= { showError }
					cleanInput 		= { cleanInput }
					onInputChange 	= { onInputChange }
				/>
				{ displayError }
			</fieldset>
		)
	}
}
Fieldset.propTypes = {
	showError: PropTypes.bool,
	cssc: PropTypes.string,
	legend: PropTypes.string,
	type: PropTypes.string,
	name: PropTypes.string,
	inputs: PropTypes.object,
	cleanInput: PropTypes.func,
	onInputChange: PropTypes.func
}

export default Fieldset
