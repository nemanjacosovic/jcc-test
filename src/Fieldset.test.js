import Fieldset from './containers/Fieldset'

 it("renders correctly", () => {
    const wrapper = shallow(
        <Fieldset
            cssc			= 'rock'
            legend 			= 'Some text'
            type   			= 'text'
            inputs 			= { {} }
            showError		= { false }
            setValue		= ''
            cleanInput 		= { () => {} }
            onInputChange	= { () => {} }
         />
    );

    expect(wrapper).toMatchSnapshot()
});