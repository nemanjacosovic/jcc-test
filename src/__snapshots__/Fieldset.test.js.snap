// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<fieldset
  className="rock"
>
  <legend>
    Some text
  </legend>
  <InputBox
    cleanInput={[Function]}
    cssc="rock"
    inputs={Object {}}
    onInputChange={[Function]}
    type="text"
  />
</fieldset>
`;
